tests/console/consoletest_setup.pm

```diff
@@ -21,7 +21,41 @@ sub run() {
 
     if (!check_var('ARCH', 's390x')) {
         # verify there is a text console on tty1
-        send_key_until_needlematch "tty1-selected", "ctrl-alt-f1", 6, 5;
+        for (1 .. 6) {
+            send_key 'ctrl-alt-f1';
+            if (check_screen("tty1-selected", 5)) {
+                last;
+            }
+        }
+        if (!check_screen "tty1-selected", 5) {    #workaround for bsc#977007
+            record_soft_failure "unable to switch to the text mode";
+            send_key 'ctrl-alt-backspace';         #kill X and log in again
+            send_key 'ctrl-alt-backspace';
+            if (get_var("NOAUTOLOGIN")) {          #copy from installation/first_boot.pm
```

Comment:
if NOAUTOLOGIN=1 is defined this testjob has autologin function disabled, we need to login to the system; if NOAUTOLOGIN has not set or 0, the testjob is auto-login to the system, like openSUSE GNOME https://openqa.opensuse.org/tests/173528/modules/first_boot/steps/1/src - it was NOAUTOLOGIN=0 and it was automatic logged-in into the system. My point is, if bsc#977007 hits openSUSE GNOME, and after ctrl-alt-backspace*2, it will be GDM after that and stop at GDM, so we have to relogin to the system like what it should be done in first_boot.pm,  otherwise openQA thinks this test is already in the desktop envirnoment after switch back to X, so it will fails to find a matched 'generic-desktop' needle at consoletest_finish.pm https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/consoletest_finish.pm#L88 , because SUT still stop at GDM at that moment.